src/python/ensembl/io/genomio/gff3/restructure.py

```diff
@@ -0,0 +1,253 @@
+        raise GFFParserError(f"Gene {gene.id} has CDSs as children in both the gene and mRNA")
+
+    # If there are exons, check they overlap with the CDSs
+    _check_sub_exons(mrna, cdss, sub_exons)
+
+    # No more issues? move the CDSs, and add any new exons
+    mrna.sub_features += cdss
+    # And remove them from the gene
+    gene.sub_features = gene_subf_clean
+    gene.sub_features.append(mrna)
+    logging.debug(f"Gene {gene.id}: moved {len(cdss)} CDSs to the mRNA")
+
+
+def _check_sub_exons(mrna: SeqFeature, cdss: SeqFeature, sub_exons: List[SeqFeature]) -> None:
+    """Check that the exons of the mRNA and the CDSs match.
+    If there are no exons, create them from the CDSs.
+    """
+
+    new_sub_exons = []
+    if sub_exons:
+        # Check that they match the CDS outside
+        if len(sub_exons) == len(cdss):
+            # Now that all coordinates are the same
+            coord_exons = [f"{exon.location}" for exon in sub_exons]
+            coord_cdss = [f"{cds.location}" for cds in cdss]
+
+            if coord_exons != coord_cdss:
+                raise GFFParserError(f"Gene CDSs and exons under the mRNA {mrna.id} do not match")
+        else:
+            raise GFFParserError(
+                f"Gene CDSs and exons under the mRNA {mrna.id} do not match (different count)"
+            )
+    else:
+        # No exons in the mRNA? Create them with the CDS coordinates
+        for cur_cds in cdss:
+            sub_exon = SeqFeature(cur_cds.location, type="exon")
+            sub_exon.sub_features = []
+            new_sub_exons.append(sub_exon)
+    mrna.sub_features += new_sub_exons
+
+
+def remove_extra_exons(gene: SeqFeature) -> None:
+    """Remove duplicated exons existing in both the gene and the mRNAs.
+
+    This is a special case where a gene contains proper mRNAs, etc. but also
+    extra exons for the same features. Those exons usually have an ID starting with
+    "id-", so that's what we use to detect them.
+    """
+    counts = _get_feat_counts(gene)
+    if not counts.get("mRNA") and not counts.get("exon"):
+        return
+
+    exons = []
+    mrnas = []
+    others = []
+    for subf in gene.sub_features:
+        if subf.type == "exon":
+            exons.append(subf)
+        elif subf.type == "mRNA":
+            mrnas.append(subf)
+        else:
+            others.append(subf)
+
+    if exons and mrnas:
+        exon_has_id = 0
+        # Check if the exon ids start with "id-", which is an indication that they do not belong here
+        for exon in exons:
+            if exon.id.startswith("id-"):
+                exon_has_id += 1
+        if exon_has_id:
+            if exon_has_id == len(exons):
+                logging.debug(f"Remove {exon_has_id} extra exons from {gene.id}")
+                gene.sub_features = mrnas
+                gene.sub_features += others
+            else:
+                raise GFFParserError(f"Can't remove extra exons for {gene.id}, not all start with 'id-'")
```

Comment:
```suggestion
        if exon_has_id == len(exons):
            logging.debug(f"Remove {exon_has_id} extra exons from {gene.id}")
            gene.sub_features = mrnas
            gene.sub_features += others
        else:
            raise GFFParserError(f"Can't remove extra exons for {gene.id}, not all start with 'id-'")
```
If you are here `len(exons)` cannot be 0, so there is no need for the first conditional check.